### PR TITLE
[patch] CP4D support cleanup

### PR DESCRIPTION
**Pull request by andrercm**

### CODE REVIEW

**Review by durera — CHANGES_REQUESTED** [review]
Instead of introducing `cpd_wd_storage_class` variable, can we put logic something like this in he default storage task:

```
# 2. Set Storage (Required)
# -----------------------------------------------------------------------------
- name: Default Metadata Storage if not set by user
  when: 
     - cpd_service_name != "wd"
     - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
  vars:
    # ROKS, OCS, Azure
    supported_storage_classes: [ibmc-file-gold-gid, ocs-storagecluster-cephfs, azurefiles-premium]
  set_fact:
    cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"

- name: Default Metadata Storage if not set by user (WD)
  when: 
     - cpd_service_name == "wd"
     - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
  vars:
    # ROKS, OCS, Azure
    supported_storage_classes: [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
  set_fact:
    cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"

- name: Assert that service storage class has been defined
  assert:
    that: cpd_service_storage_class is defined and cpd_service_storage_class != ""
    fail_msg: "cpd_service_storage_class must be defined"
```

This keeps a consistent varible name/interface regardless of the service.  All we need to do I think is change the logic for how the default is set ...

**Comment by andrercm** [comment]
> Instead of introducing `cpd_wd_storage_class` variable, can we put logic something like this in he default storage task:
> 
> ```
> # 2. Set Storage (Required)
> # -----------------------------------------------------------------------------
> - name: Default Metadata Storage if not set by user
>   when: 
>      - cpd_service_name != "wd"
>      - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
>   vars:
>     # ROKS, OCS, Azure
>     supported_storage_classes: [ibmc-file-gold-gid, ocs-storagecluster-cephfs, azurefiles-premium]
>   set_fact:
>     cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"
> 
> - name: Default Metadata Storage if not set by user (WD)
>   when: 
>      - cpd_service_name == "wd"
>      - cpd_service_storage_class is not defined or cpd_service_storage_class == ""
>   vars:
>     # ROKS, OCS, Azure
>     supported_storage_classes: [ibmc-block-gold, ocs-storagecluster-ceph-rbd, managed-premium]
>   set_fact:
>     cpd_service_storage_class: "{{ lookup_storageclasses | ibm.mas_devops.defaultStorageClass(supported_storage_classes) }}"
> 
> - name: Assert that service storage class has been defined
>   assert:
>     that: cpd_service_storage_class is defined and cpd_service_storage_class != ""
>     fail_msg: "cpd_service_storage_class must be defined"
> ```
> 
> This keeps a consistent varible name/interface regardless of the service. All we need to do I think is change the logic for how the default is set ...

This was fixed... can you help revalidate?

File: ibm/mas_devops/roles/cp4d_service/defaults/main.yml [hunk 1]
```diff
@@ -62,6 +62,9 @@ cpd_wsl_project_name: "{{ lookup('env', 'CPD_WSL_PROJECT_NAME') | default('ws_de
 cpd_wsl_project_description: "{{ lookup('env', 'CPD_WSL_PROJECT_DESCRIPTION') | default('Watson Studio Project for Maximo Application Suite', true) }}"
 cpd_wsl_project_storage_guid: "{{ 1000 | random | to_uuid | lower }}"
 
+# CPD Watson Discovery default properties
```
Comment (durera) [inline ibm/mas_devops/roles/cp4d_service/defaults/main.yml:65]: I don't think we need this nw section anymore?

File: ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml [hunk 2]
```diff
@@ -14,26 +14,37 @@
...
 - name: Assert that service storage class has been defined
   assert:
     that: cpd_service_storage_class is defined and cpd_service_storage_class != ""
     fail_msg: "cpd_service_storage_class must be defined"
 
-
 # 3. Debug storage class configuration
 # -----------------------------------------------------------------------------
 - name: "Debug CP4D service storage class configuration"
+  when: cpd_service_name != 'wd'
```
Comment (durera) [inline ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml:47]: Why don't we do the debug for discovery?

File: ibm/mas_devops/roles/cp4d_service/tasks/gencfg/gencfg-wsl.yml [hunk 3]
```diff
@@ -32,71 +32,71 @@
   set_fact:
     cpd_bearer_token: "{{ cpd_authorization.json.token }}"
 
+# Below logic to create WSL project is not needed anymore as this logic was moved to HPU operator
```
Comment (durera) [inline ibm/mas_devops/roles/cp4d_service/tasks/gencfg/gencfg-wsl.yml:35]: Has that logic been backported to all the supported MAS versions?  If not then we need logic in here to handle the versions of MAS that still need this different to the ones that don't need it.

File: ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml (hunk 2 above)
Comment (andrercm) [reply ibm/mas_devops/roles/cp4d_service/tasks/determine-storage-classes.yml:47]: fixed.

**Comment by andrercm** [comment]
@durera I'm done with the changes as part of this review... During my tests all watson services were installed correctly. Let me know if you need any further changes to this.

**Review by durera — APPROVED** [review]

**Pull request merged**